### PR TITLE
SPORTSHUB-317 Improve login/register/resetPassword UX

### PR DESCRIPTION
**Pull request by Edwin-Ni**
## Why was your change needed?

The login, register and reset password pages had poor user experience. The buttons do not show any changes when used.

## What was Changed?

Added loading status to buttons when clicked
Improved the experience.

### CODE REVIEW

File: frontend/app/(footer)/login/page.tsx [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
-        router.push("/dashboard?login=success"); // Redirect only if user creation is successful
-      } else {
-        console.error("User creation failed.");
+    startTransition(async () => {
+      try {
+        const userCreated = await handleEmailAndPasswordSignIn(userData.email, userData.password);
+        if (userCreated) {
+          router.push("/dashboard?login=success"); // Redirect only if user creation is successful
+        } else {
+          console.error("User creation failed.");
```
Comment (brian2w) [inline frontend/app/(footer)/login/page.tsx:30]: Should we be alerting the user if their account creation has failed? This currently only logs the error to the console.

**Review by brian2w — APPROVED** [review]
NICE!

File: frontend/app/(footer)/login/page.tsx (hunk 1 above)
Comment (Edwin-Ni) [reply frontend/app/(footer)/login/page.tsx]: Good catch, changed it to throw and catch the error instead

**Review by brian2w — APPROVED** [review]
lgtm

**Pull request merged**